metaflow/plugins/argo/argo_workflows_cli.py

```diff
@@ -385,48 +511,46 @@ def resolve_workflow_name(obj, name):
             % to_unicode(base64.b32encode(sha1(project_branch).digest()))[:16]
         )
         is_project = True
-        # Argo Workflow names can't be longer than 253 characters, so we truncate
-        # by default. Also, while project and branch allow for underscores, Argo
-        # Workflows doesn't (DNS Subdomain names as defined in RFC 1123) - so we will
-        # remove any underscores as well as convert the name to lower case.
-        # Also remove + and @ as not allowed characters, which can be part of the
-        # project branch due to using email addresses as user names.
-        if len(workflow_name) > 253:
+        if len(workflow_name) > limit:
             name_hash = to_unicode(
                 base64.b32encode(sha1(to_bytes(workflow_name)).digest())
-            )[:8].lower()
-            workflow_name = "%s-%s" % (workflow_name[:242], name_hash)
-            obj._is_workflow_name_modified = True
+            )[:5].lower()
+            # TODO: We can create better names that try to preserve the flow name
+            workflow_name = "%s-%s" % (workflow_name[: limit - 6], name_hash)
```

Comment:
we could use `current.flow_name` as the base and append the hash which comes from the `current.project_flow_name` if the flow name is the most significant info in the template name